File: Brece-cs50-2018-spring-cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float n;
    int coins = 0;
    int quarter = 0;
    int dime = 0;
    int nickel = 0;
    int penny = 0;

    do
    {
        n = get_float("Change owed: ");
    }
    while (n < 0);

    //convert float to int
    int cash = (int)(100.0 *n);

    while (cash >= 25)
    {
        cash -= 25;
        coins += 1;
        quarter += 1;
    }

    while (cash >= 10)
    {
        cash -= 10;
        coins += 1;
        dime += 1;
    }
    
    while (cash >= 5)
    {
        cash -= 5;
        coins += 1;
        nickel += 1;
    }

    while (cash >= 1)
    {
        cash -= 1;
        coins += 1;
        penny += 1;
        
    }
    
    
    printf("Quarter: %i ; dime: %i ; nickel: %i ; penny: %i\n", quarter, dime, nickel, penny);
    printf("Coins: %i\n", coins);
}

File: Brece-cs50-2018-x-sentimental-cash/cash.py
import cs50

while True:
    print("Change owed: ", end="")
    n = cs50.get_float()
    if n > 0:
        break
    

coins = 0
quarter = 0
dime = 0
nickel = 0
penny = 0

cash = int(100.0 * n)

if cash >= 25:
    coins += (cash//25)
    quarter += (cash//25)
    cash -= 25*(cash//25)
    
if (cash%25) >= 10:
    coins += (cash//10)
    dime += (cash//10)
    cash -= 10*(cash//10)
    
if (cash%10) >= 5:
    coins += (cash//5)
    nickel += (cash//5)
    cash -= 5*(cash//5)
    
if (cash%5) < 5:
    coins += (cash%5)
    penny += (cash%5)
    cash -= (cash%5)
    
print("Coins: " + str(coins))
print("Quarter: " + str(quarter))
print("Dime: " + str(dime))
print("Nickel: " + str(nickel))
print("Penny: " + str(penny))
print("Cash owed: " + str(cash))